### PR TITLE
Cap broadcast start shell

### LINKED ISSUE #68292: Bluetooth: LE Audio: CAP Commander: Broadcast Reception Start procedure shell commands
**Is your feature request related to a problem? Please describe.**
Implement a shell command to use the Broadcast Reception Start procedure mentioned in issue #64358 

**Describe the solution you'd like**
Implement a shell command consistent with the existing shell commands

**Describe alternatives you've considered**
N/A

**Additional context**
See #64358

### PR DESCRIPTION
**Pull request by kruithofa**
Added the shell command for the new broadcast reception start procedure

fixes #68292

### CODE REVIEW

File: subsys/bluetooth/audio/shell/cap_commander.c [hunk 1]
```diff
@@ -472,6 +486,173 @@ static int cmd_cap_commander_change_microphone_gain(const struct shell *sh, size
...
+	member_param->subgroups = &subgroup;
+
+	member_param->member.member = connected_conns[0];
+
+	for (size_t i = 1U; i < conn_cnt; i++) {
+		memcpy(&member_params[i], member_param, sizeof(*member_param));
+
+		/* the member value is different for each, so we can not just copy this value */
+		member_params[i].member.member = connected_conns[i];
+	}
```
Comment (Thalley) [inline subsys/bluetooth/audio/shell/cap_commander.c:639]: Suggest to add a comment on why we are copying the parameters

**Review by Thalley — CHANGES_REQUESTED** [review]
Minor request, otherwise LGTM

**Review by Thalley — DISMISSED** [review]

File: subsys/bluetooth/audio/shell/cap_commander.c [hunk 2]
```diff
@@ -472,6 +486,177 @@ static int cmd_cap_commander_change_microphone_gain(const struct shell *sh, size
...
+
+	param.count = conn_cnt;
+
+	shell_print(sh, "Starting broadcast on %zu connections", param.count);
```
Comment (Thalley) [inline subsys/bluetooth/audio/shell/cap_commander.c:647]:
```suggestion
	shell_print(sh, "Starting broadcast reception on %zu connections", param.count);
```

File: doc/connectivity/bluetooth/api/audio/shell/cap.rst [hunk 3]
```diff
@@ -436,3 +436,18 @@ and index 1 gets the second offset, etc.:
...
+
+   uart:~$ bt connect <device A>
+   Connected: <device A>
+   uart:~$ bap_init
+   uart:~$ cap_commander discover
+   discovery completed with CSIS
+   uart:~$ bap_broadcast_assistant discover
+   BASS discover done with 1 recv states
+   uart:~$ cap_commander broadcast_reception_start <device B> 0 4
+   Starting broadcast on 1 connections
```
Comment (Thalley) [inline doc/connectivity/bluetooth/api/audio/shell/cap.rst:453]:
```suggestion
   Starting broadcast reception on 1 connections
```

File: subsys/bluetooth/audio/shell/cap_commander.c [hunk 4]
```diff
@@ -87,6 +87,17 @@ static void cap_microphone_gain_changed_cb(struct bt_conn *conn, int err)
...
 #endif /* CONFIG_BT_MICP_MIC_CTLR */
 
+#if defined(CONFIG_BT_BAP_BROADCAST_ASSISTANT)
+static void cap_broadcast_reception_start_cb(struct bt_conn *conn, int err)
+{
+	if (err != 0) {
+		shell_error(ctx_shell, "Broadcap reception start failed (%d)", err);
+		return;
+	}
+	shell_print(ctx_shell, "Broadcap reception start completed");
```
Comment (Thalley) [inline subsys/bluetooth/audio/shell/cap_commander.c:97]:
```suggestion
	}
	
	shell_print(ctx_shell, "Broadcast reception start completed");
```

File: subsys/bluetooth/audio/shell/cap_commander.c [hunk 5]
```diff
@@ -87,6 +87,17 @@ static void cap_microphone_gain_changed_cb(struct bt_conn *conn, int err)
 #endif /* CONFIG_BT_MICP_MIC_CTLR_AICS */
 #endif /* CONFIG_BT_MICP_MIC_CTLR */
 
+#if defined(CONFIG_BT_BAP_BROADCAST_ASSISTANT)
+static void cap_broadcast_reception_start_cb(struct bt_conn *conn, int err)
+{
+	if (err != 0) {
+		shell_error(ctx_shell, "Broadcap reception start failed (%d)", err);
```
Comment (Thalley) [inline subsys/bluetooth/audio/shell/cap_commander.c:94]:
```suggestion
		shell_error(ctx_shell, "Broadcast reception start failed (%d)", err);
```

**Review by Thalley — CHANGES_REQUESTED** [review]

**Review by Thalley — APPROVED** [review]

File: doc/connectivity/bluetooth/api/audio/shell/cap.rst [hunk 6]
```diff
@@ -436,3 +436,18 @@ and index 1 gets the second offset, etc.:
...
+Starting a broadcast reception
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: console
+
+   uart:~$ bt connect <device A>
+   Connected: <device A>
+   uart:~$ bap_init
+   uart:~$ cap_commander discover
+   discovery completed with CSIS
```
Comment (larsgk) [inline doc/connectivity/bluetooth/api/audio/shell/cap.rst:449]: Can we do this with more than 1 device? If so, maybe a good example

**Review by larsgk — APPROVED** [review]

**Pull request merged**